src/main/java/bimo/Parser.java

```diff
@@ -0,0 +1,155 @@
+import bimo.command.ListCommand;
+import bimo.command.MarkCommand;
+import bimo.command.UnmarkCommand;
+import bimo.tasks.Deadline;
+import bimo.tasks.Event;
+import bimo.tasks.Task;
+import bimo.tasks.ToDo;
+
+
+
+
+/**
+ * Deals with making sense of the user command.
+ */
+public class Parser {
+
+    /**
+     * Converts user input to the appropriate Command object.
+     *
+     * @param input Input that user typed in.
+     * @return Command object.
+     * @throws BimoException If there is an invalid command or invalid date.
+     */
+    public static Command parse(String input) throws BimoException {
+        String[] parsedArray = input.split(" ");
+        String cmd = parsedArray[0].toLowerCase();
+        if (cmd.equals("list")) {
+            return new ListCommand();
+
+        } else if (cmd.equals("mark")) {
+            int index = Integer.valueOf(parsedArray[1]) - 1;
+            return new MarkCommand(index);
+
+        } else if (cmd.equals("unmark")) {
+            int index = Integer.valueOf(parsedArray[1]) - 1;
+            return new UnmarkCommand(index);
+
+        } else if (cmd.equals("todo")) {
+            String description = parseDescription(input);
+            return new AddCommand(new ToDo(description));
+
+        } else if (cmd.equals("event")) {
+            String[] array = input.split("/from ");
+            String description = parseDescription(array[0]);
+            String start = parseDate(false, false, array);
+            String end = parseDate(false, true, array);
+            LocalDate startDate = convertDate(start);
+            LocalDate endDate = convertDate(end);
+            Task task = new Event(description, startDate,
+                   endDate);
+            return new AddCommand(task);
+
+        } else if (cmd.equals("deadline")) {
+            String[] array = input.split("/by ");
+            String description = parseDescription(array[0]);
+            String dueDate = parseDate(true, true, array);
+            LocalDate ld = convertDate(dueDate);
+            Task task = new Deadline(description, ld);
+            return new AddCommand(task);
+
+        } else if (cmd.equals("delete")) {
+            int index = Integer.valueOf(parsedArray[1]) - 1;
+            return new DeleteCommand(index);
+
+        } else if (cmd.equals("bye")) {
+            return new ByeCommand();
+        } else if (cmd.equals("find")) {
+            String[] words = input.split(" ");
+            return new FindCommand(words);
+
+        } else {
+            throw new BimoException("Sorry, I do not understand you \n"
+                    + "as this is not a valid command");
+        }
+    }
+
+    /**
+     * Returns description portion of user input.
+     *
+     * @param input Input typed in by user.
+     * @return String value of task description.
+     * @throws BimoException If there is no description typed in.
+     */
+    public static String parseDescription(String input) throws BimoException {
```

Comment:
I really like that you abstracted out these methods! It makes it much easier to read the code. The naming of the methods are also very clear.